### ISSUE TITLE
PATH too long

### ISSUE THREAD

**Issue by mycall** [pothosware/PothosSDR#46]
I've had success in using this PowerShell script to pack the PATH environment variable, in order to support more directories, and thought I would share it.  

clear
write-host "Reading..."
write-host

$fso = New-Object -ComObject "Scripting.FileSystemObject"
$shortpaths = @();
$originalPaths = [environment]::GetEnvironmentVariable("path", "Machine").Split(";")

foreach ($path in $originalPaths) {
	$fpath = [System.IO.Path]::GetFullPath("$path");
    $fspath = $fso.GetFolder("$fpath").ShortPath;
	$foundIdx = $shortpaths.IndexOf($fspath);
	if ($foundIdx -gt -1) {	continue; }
	write-host $fpath  -->  $fspath;
	$shortpaths += $fspath;
}
write-host
write-host "Packing..."
write-host
$env:Path = $($shortpaths -join ";");
write-host $env:Path
write-host
write-host "Done." 
write-host

**Comment by gvanem** [comment]
Maybe related, but I get the below *Warning* using this installer: `PothosSDR-2017.11.26-vc14-x64.exe`:

![pothos-2017-install](https://user-images.githubusercontent.com/945271/33997527-f9e846f0-e0e4-11e7-920a-c97d968b5e86.png)

**Comment by nlhnt** [comment]
Any news on that?
I got the same error during the installation.
When I manually added C:\Program Files\PothosSDR\bin to my PATH variable it kind of turned Pothos Flow on, but all the GNURadio blocks I put down are read.

**Comment by TomCox01** [comment]
Use Long Path Tool, Long Path Tool can simplify and 

probably end your problems in unlocking, managing and 

renaming files that appear to have a long filename.

**Comment by einthecorgi2** [comment]
Same problem

**Comment by vbursucianu** [comment]
Got this issue too

**Comment by n3rdx** [comment]
yes, I installed the latest I think (PothosSDR-2021.01.28-vc16-x64) today and got the same scary error. I will attempt to re-install it with a slightly shorter (but by how much) pathname. Just thought I would record this on Windows 10. I can install all commercial programs ... UPDATE: I de-installed and then re-installed with a C:\Program Files\PSDR directory location, (shorter path name) and so far it hasn't complained and just finished installing. However the de-installation process (UNINSTALL) was not smooth, and I had to restart the computer/use CCleaner/clean registry before I could get the entry removed from the Windows app registry which then would allow me to install once again. Otherwise, I was getting a notice that the App was already installed, thus it couldn't be installed again, even if I chose to install over the current installation (which was empty BTW). Bug?

**Comment by guruofquality** [comment]
I wonder if this will fix it: https://docs.python.org/3/using/windows.html#removing-the-max-path-limitation

> Windows historically has limited path lengths to 260 characters. This meant that paths longer than this would not resolve and errors would result.
> In the latest versions of Windows, this limitation can be expanded to approximately 32,000 characters. Your administrator will need to activate the “Enable Win32 long paths” group policy, or set LongPathsEnabled to 1 in the registry key HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\FileSystem.

**Comment by Skinner927** [comment]
FWIW, enabling long paths did not change anything.
I have tried installing in `c:\psdr`, but I still get the path error.

It seems to have something to do with adding user or system's `PATH` environment variable rather than file path because if I turn that setting off in the installer, the error goes away.

**Comment by ratzrattillo** [comment]
Experienced the same issue. Is this being worked on?
Please inform us, if there is a workaround, or how this is planned to be fixed!

**Comment by habub** [comment]
This issue was opened in 2017, now it's 2022, I also have this same issue (2021.07.25-vc16-x64).
Any resolution I missed?

**Comment by nikstp** [comment]
I also got that error during the first install. I've had no issues (re)installing the software without adding the folder to the path automatically and instead adding it manually; I'd count that as a workaround for now.

**Comment by Jitteryheart87** [comment]
how do i add this manually?
